### PR TITLE
Locate User Directory without $HOME env var via getpwnam_r.

### PR DESCRIPTION
**Pull request by UsairimIsani**
This pull request eliminates the need to check the $HOME env variable.

More pull request to come if accepted.

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -22,7 +22,8 @@
 #![allow(dead_code)] // This module acts more like a library, so not yet used is ok.
 
 use log::debug;
-use std::env;
+use nix::unistd::{Uid, User};
+use std::ffi::OsString;
```
Comment (rusty-snake) [inline src/utils.rs:26]: All other use statements in this file `use`s the module (exception: log macros and traits) and `use`s the types only in the function body.

Comment (UsairimIsani) [reply src/utils.rs:26]: Okay,  Noted.
Should this convention be followed througout the project?

Comment (rusty-snake) [reply src/utils.rs:26]: Only in files where "other"-function/types/traits/… are. Currently `macros/src/lib.rs` and `src/utils.rs`.

**Comment by UsairimIsani** [comment]
I have made a mistake, getpwnam_r is not a syscall, indicated by syscalls man page.

**Comment by UsairimIsani** [comment]
Using libc would require me to use unsafe foreign functions or make safe wrappers on those functions. Nix provides those safe wrappers. What would you prefer?

**Comment by rusty-snake** [comment]
After some thinking, I think in the long run it is easier to go with nix.

BTW: Its' `getpwuid_r` https://docs.rs/nix/0.17.0/nix/unistd/struct.User.html#method.from_uid.

File: Cargo.toml [hunk 2]
```diff
@@ -21,6 +21,7 @@ bitflags = "1.2"
 lazy_static = "1"
 log = "0.4"
 termcolor = "1.1"
+nix = "0.17.0"
```
Comment (rusty-snake) [inline Cargo.toml:24]: Notes on my/fjps deps specify rules:
- I usually omit the patch version.
- If features are specified, the `[dependencies.foo]` syntax is used.
- They are order alphabetically.
- Ordering: `[dependencies]`; `[dependencies.foo]`; `[dependencies.bar]` (optional = true); path; build; dev

File: src/utils.rs [hunk 3]
```diff
@@ -111,19 +112,20 @@ pub fn get_name1(raw: &str) -> String {
...
-        .map(PathBuf::from)
+    use ffi::OsString;
+    use unistd::{Uid, User};
+    let user = User::from_uid(Uid::current()).unwrap().unwrap();
+    let dir = OsString::from(user.dir);
+    if dir.len() > 0 {
+        Some(path::PathBuf::from(dir))
+    } else {
+        None
+    }
```
Comment (rusty-snake) [inline src/utils.rs:128]: `user.dir` is already a PathBuf. Can we simplify this to something like
```rust
let path = user.dir;
if path.as_os_str().is_empty() {
```

**Comment by UsairimIsani** [comment]
I have updated the contributing.md to reflect current conventions used.
I also have incorporated your suggestions in the home_dir code.

**Comment by rusty-snake** [comment]
Merged, Thanks.

**Pull request merged**